File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/interfaceInJava/interfaceExample/Vehicle.java
package de.fraunhofer.iem.classesAndObjects.interfaceInJava.interfaceExample;

/**
 * Demonstrates the Interface in Java
 *
 * @author Ranjith Krishnamurthy
 */
public interface Vehicle {
    public void start();

    public void stop();

    public void accelerate(int speed);

    public void changeGear(int to);

    public int getCurrentSpeed();
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/interfaceInKotlin/interfaceExample/Vehicle.kt
package de.fraunhofer.iem.classesAndObjects.interfaceInKotlin.interfaceExample

/**
 * Demonstrates the Interface in Kotlin
 *
 * @author Ranjith Krishnamurthy
 */
interface Vehicle {
    fun start()
    fun stop()
    fun accelerate(speed: Int)
    fun changeGear(to: Int)
    fun getCurrentSpeed(): Int
}